modal/object.py

```diff
@@ -44,6 +44,8 @@ class _Object:
     _object_id: str
     _client: _Client
     _is_hydrated: bool
+    _is_rehydrating: bool
+    _is_rehydrated: bool
```

Comment:
Is there a reason we need separate `_is_rehydrating` and `_is_rehydrated` attributes rather than just toggling `_is_hydrated` to `False` on restore?